src/client.ts

```diff
@@ -355,6 +364,73 @@ export class Client {
     }
   }
 
+  /**
+ * Take a screenshot of the page and save it to `filename` in `path` folder, with a `format` and `quality` (jpeg format only)
+ * If `selector` is passed in, it will take a screenshot of only that element
+ * and its children as opposed to the whole page.
+ *
+ * @param path - The path of where to save the screenshot to
+ * @param options - options
+ * @param options.filename - Name to be given to the screenshot. Optional
+ * @param options.selector - Screenshot the given selector instead of the full page. Optional
+ * @param options.format - The Screenshot format(and hence extension). Allowed values are "jpeg", "png", and "webp" - Optional
+ * @param options.quality - The image quality from 0 to 100, default 80. Applicable only if no format provided or format is "jpeg" - Optional
+ */
+  public async takeScreenshot(
+    path: string,
+    options?: {
+      selector?: string;
+      fileName?: string;
+      format?: "jpeg" | "png" | "webp";
+      quality?: number;
+    },
+  ): Promise<string> {
+    if (!existsSync(path)) {
+      await this.done();
+      throw new Error(`The provided folder path - ${path} doesn't exist`);
+    }
+    const ext = options?.format ?? "jpeg";
+    // deno-lint-ignore ban-types
+    const clip: Object | undefined = (options?.selector)
+      ? await this.getViewport(options.selector)
+      : undefined;
+
+    if (options?.quality && Math.abs(options.quality) > 100 && ext == "jpeg") {
+      await this.done();
+      throw new Error("A quality value greater than 100 is not allowed.");
+    }
+
+    //Quality should defined only if format is jpeg
+    const quality = (ext == "jpeg")
+      ? ((options?.quality) ? Math.abs(options.quality) : 80)
+      : undefined;
+
+    const res = await this.sendWebSocketMessage(
+      "Page.captureScreenshot",
+      {
+        format: ext,
+        quality: quality,
+        clip: clip,
+      },
+    ) as {
+      data: string;
+    };
+
+    //Writing the Obtained Base64 encoded string to image file
+    const fName =
```

Comment:
Just FYI I believe the right hand side of the ?? operator will never be called because the left hand side will always be truths (as path is already a truths value) right?

Reply:
No, the tests work as expected, timestamp is there if filename not provided...

Reply:
Because I'm using ${} so path is not checked as it is outside